src/main/java/pl/pandziak/ticketbookingapp/domain/reservation/dtos/ReservationRequestDto.java

```diff
@@ -0,0 +1,109 @@
+package pl.pandziak.ticketbookingapp.domain.reservation.dtos;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.lang3.StringUtils;
+import pl.pandziak.ticketbookingapp.domain.TicketPricing;
+import pl.pandziak.ticketbookingapp.domain.reservation.Reservation;
+import pl.pandziak.ticketbookingapp.domain.reservation.Ticket;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.toList;
+
+public class ReservationRequestDto {
+
+    private final String screeningId;
+    private final String userName;
+    private final String userSurname;
+    private final TicketsDto tickets;
+
+    @JsonCreator
+    public ReservationRequestDto(@JsonProperty String screeningId,
+                          @JsonProperty String userName,
+                          @JsonProperty String userSurname,
+                          @JsonProperty TicketsDto tickets) {
+        this.screeningId = screeningId;
+        this.userName = userName;
+        this.userSurname = userSurname;
+        this.tickets = tickets;
+    }
+
+    private ReservationRequestDto(Builder builder) {
+        this(
+                builder.screeningId,
+                builder.userName,
+                builder.userSurname,
+                builder.tickets);
+    }
+
+    public boolean isValid() {
+        return StringUtils.isNoneBlank(userName, userSurname)
+                && tickets != null;
```

Comment:
To jest nienajlepszy warunek, wiesz dlaczego?

Reply:
Chodzi o niesprawdzenie czy 'tickets' nie jest pustą tablicą? Wydawało mi się to bardziej założenie domenowe, że przy rezerwacji musi być wybrany chociaż jeden bilet. I taka walidacja odbywa się w późniejszym etapie, a metoda 'isValid' sprawdza tylko input użytkownika pod kątem wypełnienia pól.

Reply:
Właśnie - że musi być jeden bilet. A co jeśli w szaleństwie refactoru ktoś zrobi tak, że to nigdy nie jest nullem, tylko pustą kolekcją? Łatwo zmienić tam, a zapomnieć tutaj.

Reply:
Ok, słuszna uwaga. :)